src/db/couchDevices.ts

```diff
@@ -46,6 +47,19 @@ export const asCouchDevice = asCouchDoc(
     deviceToken: asOptional(asString),
     ignoreMarketing: asOptional(asBoolean, false),
     ignorePriceChanges: asOptional(asBoolean, false),
+    ip: asOptional(asString, 'unknown'),
+    location: asOptional(
+      asEither(
+        asObject({
+          country: asString,
+          city: asString,
+          region: asOptional(asString, '')
+        }),
+        asObject({
+          error: asString
+        })
```

Comment:
Does it make sense to store errors? If we fail to look up a row, this could just be missing, as it is for old rows where we don't have it either. If the API goes down, I would expect to see that in the logs, as opposed to the database.

Reply:
If the IP address fails to get a location consistently, I didn't want the user's location to be stuck on some old IP's location. I suppose, we could just set the location to undefined in this case. I'll do that instead. 

Also, should we locate the IP on each call or only when the IP has changed? I suppose if IP locations don't change often, then we should be safe to assume the IP location is a constant and can optimize here.